core/trino-parser/src/main/antlr4/io/trino/sql/parser/SqlBase.g4

```diff
@@ -418,6 +418,54 @@ relationPrimary
     | LATERAL '(' query ')'                                           #lateral
     | TABLE '(' tableFunctionCall ')'                                 #tableFunctionInvocation
     | '(' relation ')'                                                #parenthesizedRelation
+    | JSON_TABLE '('
+        jsonPathInvocation
+        COLUMNS '(' jsonTableColumn (',' jsonTableColumn)* ')'
+        (PLAN '(' jsonTableSpecificPlan ')'
+        | PLAN DEFAULT '(' jsonTablePlanDefault ')')?
+        ((ERROR | EMPTY) ON ERROR)?
+      ')'                                                             #jsonTable
+    ;
+
+jsonTableColumn
+    : identifier FOR ORDINALITY                                     #ordinalityColumn
+    | identifier type
+        (PATH string)?
+        (emptyBehavior=jsonValueBehavior ON EMPTY)?
+        (errorBehavior=jsonValueBehavior ON ERROR)?                 #valueColumn
+    | identifier type FORMAT jsonRepresentation
+        (PATH string)?
+        (jsonQueryWrapperBehavior WRAPPER)?
+        ((KEEP | OMIT) QUOTES (ON SCALAR TEXT_STRING)?)?
+        (emptyBehavior=jsonQueryBehavior ON EMPTY)?
+        (errorBehavior=jsonQueryBehavior ON ERROR)?                 #queryColumn
+    | NESTED PATH? string (AS identifier)?
+        COLUMNS '(' jsonTableColumn (',' jsonTableColumn)* ')'      #nestedColumns
+    ;
+
+jsonTableSpecificPlan
+    : leafPlan
+    | compositePlan
+    ;
+
+leafPlan
+    : identifier
+    ;
+
+compositePlan
+    : leafPlan (OUTER | INNER) planPrimary                      #planParentChild
+    | planPrimary UNION planPrimary (UNION planPrimary)*        #planSiblingsUnion
+    | planPrimary CROSS planPrimary (CROSS planPrimary)*        #planSiblingsCross
```

Comment:
Why the distinction between "parent-child" and "siblings" in the labels? That seems like a semantic difference, not something relevant to the grammar.

It would be more appropriate to use `joinPlan`, `unionPlan`, and `crossPlan`, instead, IMO.